### DISCUSSION TITLE
503 (Service Unavailable) - How Often Does This Happen To The NWS API?

### DISCUSSION THREAD

**Discussion by jkreski** [weather-gov/api#730] [General]
I've seen this in both of the last two days.  Wondering how reliable the API actually is...

**Comment by jah1007** [comment]
URL or it isn’t happening 🙃

**Reply by jkreski** [reply]
https://api.weather.gov/gridpoints/APX/14,39/forecast

**Reply by jah1007** [reply]
That site has had communications issues the last two days which explains why their forecasts have been spotty. If the forecast is older than 15 hours it throws a 503 because it is expired.

**Reply by jkreski** [reply]
Thanks!  Would that affect current temperature on the web vs API?  They have been wonky...

**Reply by scadergit** [reply]
The forecast data is provided the by field office, but current observations are provided by MADIS. There has been issues with MADIS data lately, which is being investigated separately.

**Comment by brewin** [comment]
I’ve found that retrying 5 times with a delay of 1 second between retries sometimes works. It’s not ideal and I’m not sure what to do when that doesn’t work. Maybe attempt to get the forecast for another nearby grid point.

**Reply by scadergit** [reply]
This type of behavior is a separate issue with the forecast data where the upstream service from the API that transcodes the weather sent by the office can intermittently fail. It's currently being addressed.

**Reply by brewin** [reply]
Thanks. I'm looking forward to these issues being resolved.
